src/Compilers/Core/CodeAnalysisTest/DefaultAnalyzerAssemblyLoaderTests.cs

```diff
@@ -127,6 +127,48 @@ public void Exec(ITestOutputHelper testOutputHelper, bool shadowLoad, string typ
 
 #endif
 
+    /// <summary>
+    /// Contains the bulk of our analyzer / generator loading tests.
+    /// </summary>
+    /// <remarks>
+    /// These tests often have quirks associated with fundamental limitation issues around either 
+    /// .NET Framework, .NET Core or our own legacy decisions. Rather than repeating a specific rationale
+    /// at all the tests that hit them, the common are outlined below and referenced with the following 
+    /// comment style within the test.
+    ///
+    ///    // See limitation 1
+    ///
+    /// This allows us to provide central description of the limitations that can be easily referenced in the impacted
+    /// tests. For all the descriptions below assume that A.dll depends on B.dll. 
+    ///
+    /// Limitation 1: .NET Framework probing path.
+    ///
+    /// The .NET Framework assembly loader will only call AppDomain.AssemblyResolve when it cannot satifisfy a load
+    /// request. One of the places the assembly loader will always consider when looking for dependecies of A.dll
+    /// is the directory that A.dll was loading from (it's added to the probing path). Than means if B.dll is in the
+    /// same directory then the runtime will silently load it without a way for us to intervene.
+    ///
+    /// Note: this only applies when A.dll is in the Load or LoadFrom context which is always true for these tests
+    /// 
+    /// Limitation 2: Dependency is already loaded.
+    ///
+    /// Similar to Limitation 1 is when the dependency, B.dll, is already present in the Load or LoadFrom context 
+    /// then that will be used. The runtime will not attempt to load a better version (an exact match for example).
+    ///
+    /// Limitation 3: Shadow copy breaks up directories
```

Comment:
I expected limitation 3 to be referenced in some of the below tests, but it's not urgent to make the change to do so.

Reply:
It is referenced in `AssemblyLoading_DependencyInDifferentDirectory2`.